docs/user-documentation/moderne-platform/how-to-guides/writing-and-installing-recipes.md

```diff
@@ -0,0 +1,38 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Writing and Installing Recipes
+
+[OpenRewrite](https://docs.openrewrite.org/) is a framework for writing and running code transformations. A new code transformation, called [recipes](https://docs.openrewrite.org/concepts-and-explanations/recipes) can be written
+and uploaded to your instance of the Moderne Platform. This is done in 3 steps:
+
+1. Writing custom recipes
+2. Publish artifact containing the custom recipes
+3. Importing Recipes
+
+## Writing Recipes
+Before writing a new recipe, check the [recipe catalogue](https://docs.openrewrite.org/recipes) to see if an existing recipe is available. For transformations that aren't covered by the existing catalogue, new recipes can be written in:
+ - [Declarative YAML](https://docs.openrewrite.org/concepts-and-explanations/recipes#declarative-recipes)
+ - [Refaster](https://docs.openrewrite.org/authoring-recipes/refaster-recipes)
+ - [Imperative Java](https://docs.openrewrite.org/concepts-and-explanations/recipes#imperative-recipes).
+
+As a best practice, if your recipe can be built out of other recipes, you should make it declarative. You can make some powerful migration recipes by
+combining many tiny recipes. Depending on the code transformation a declarative YAML recipe might not be possible, and using an imperative Java or a 
+Refaster recipe would be more appropriate. For details on how to write recipes follow these [instructions](https://docs.openrewrite.org/authoring-recipes/types-of-recipes). Try our [Recipe authoring workshop](../../workshops/recipe-authoring.md) 
+for a step-by-step example of recipe authoring.
+
+## Creating a Recipe Jar
+
+After the new recipes have been written, they need to be published to an Artifactory which has been set up as a [recipe source](../../../administrator-documentation/moderne-platform/how-to-guides/agent-configuration/configure-an-agent-with-maven-repository-access.md#publishing-recipe-artifacts).
```

Comment:
It doesn't have to be an Artifactory, it can be any Maven repository.

Also, we need to consider (cc @macsux @mike-solomon ) how to start writing docs for writing non-Java recipes.

Reply:
I think the latests version of this file already addressed this :) 

The latest version of the file has changed since this PR, this sentence is now `you will need to publish them to an artifact repository that is connected to your Moderne instance`.